onnx/onnx_pb.h

```diff
@@ -8,14 +8,9 @@
 // different declarations (dllexport and dllimport). On Linux/Mac, it just
 // resolves to the same "default visibility" setting.
 #if defined(_MSC_VER)
-#if defined(ONNX_BUILD_SHARED_LIBS) || defined(ONNX_BUILD_MAIN_LIB)
```

Comment:
Why do you remove this line?
What if people want to build a static lib?
What if people want to consume this header and static link to onnx?

Reply:
Thanks for your review. This line is not removed accidentally in the commit of removing ONNX_BUILD_MAIN_LIB.

The dllexport and dllimport are now managed by ONNX_EXPORT_SYMBOLS and ONNX_IMPORT_SYMBOLS, which are set properly in cmake.

The reason to change this line is to extract the common codes. For example, in the original code, there are two "ONNX_BUILD_MAIN_LIB", one is [a cmake variable](https://github.com/onnx/onnx/blob/master/CMakeLists.txt#L306) and another is [a compiler definition](https://github.com/onnx/onnx/blob/master/onnx/onnx_pb.h#L11). Setting the cmake variable to ON doesn't define its namesake. How can users define this compiler definition? No easy way. I believe it is not the expected behavior. This bug clearly shows that it is error-prone to maintain the duplicated logic in two different places. So I extract the logic to cmake, and set the two extra definitions to control the behavior of onnx_pb.h.

Reply:
Yes, it's a bug. It can be easily fixed by adding:
```
if(ONNX_BUILD_MAIN_LIB)
  target_compile_definitions(xxx  PRIVATE ONNX_BUILD_MAIN_LIB)
endif()
```
to cmake files.

Reply:
I personally prefer this idea of removing the duplicated logic in both header and cmake files. :)